src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationService.kt

```diff
@@ -192,6 +192,40 @@ class ApplicationService(
     )
   }
 
+  @SuppressWarnings("ReturnCount")
+  fun abandonApplication(applicationId: UUID, user: NomisUserEntity):
+    AuthorisableActionResult<ValidatableActionResult<Cas2ApplicationEntity>> {
+    val application = applicationRepository.findByIdOrNull(applicationId)
+      ?: return AuthorisableActionResult.NotFound()
+
+    if (application.createdByUser != user) {
+      return AuthorisableActionResult.Unauthorised()
+    }
+
+    if (application.submittedAt != null) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.ConflictError(applicationId, "This application has already been submitted"),
+      )
+    }
+
+    if (application.abandonedAt != null) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.Success(application),
+      )
+    }
+
+    application.apply {
+      this.abandonedAt = OffsetDateTime.now()
+      this.data = null
```

Comment:
Just checking that this won't break the unsubmitted application report? If I remember rightly we should have pulled everything we need out into first class columns, avoiding any json digging?

Reply:
hmm it doesn't break the report but it doesn't filter out the abandoned applications from the report...that would [have to happen here](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/b7d5dec5f0b7ef046d14ec9cc5205c3d6fc10cf4/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2UnsubmittedApplicationsReport.kt).

To avoid holding this up, shall I make a new subtask on this ticket and handle any MI logic in a new PR?
Do we want them removed from the MI report, or do we want them included but with a column for whether they were withdrawn?

Reply:
[As mentioned on Slack](https://mojdt.slack.com/archives/C04T6SC31HA/p1717064275466769?thread_ts=1717063601.939889&cid=C04T6SC31HA) my thoughts are in agreement that we keep them in the submitted report and create a follow-up task to add a column which shares the abandoned state with Suzie 👍